mbed/source/main.cpp

```diff
@@ -0,0 +1,423 @@
+        if (error) {
+            print_error(error, "Gap::startScan failed");
+            return;
+        }
+
+        printf("Scanning for %ums\r\n", _scan_time.valueInMs());
+    }
+
+    void connect_peripheral()
+    {
+        update_state(bt_test_state_t::CONNECT_PERIPHERAL);
+        printf("Connected as peripheral\r\n");
+    }
+
+    void connect_master()
+    {
+        update_state(bt_test_state_t::CONNECT_MASTER);
+        printf("Connected as master\r\n");
+    }
+
+private:
+    /* Gap::EventHandler */
+
+    void onAdvertisingStart(const ble::AdvertisingStartEvent &event) override
+    {
+        if (_is_periodic) {
+            ble_error_t error = _ble.gap().setPeriodicAdvertisingParameters(
+                _adv_handle,
+                ble::periodic_interval_t(100),
+                ble::periodic_interval_t(1000)
+            );
+            if (error) {
+                print_error(error, "Gap::setPeriodicAdvertisingParameters() failed");
+                return;
+            }
+
+            error = _ble.gap().startPeriodicAdvertising(_adv_handle);
+            if (error) {
+                print_error(error, "Gap::startPeriodicAdvertising() failed");
+                return;
+            }
+
+            printf("Periodic advertising for %ums\r\n", _advertise_time.valueInMs());
+        } else {
+            printf("Advertising started for %ums\r\n", _advertise_time.valueInMs());
+        }
+    }
+
+    void onAdvertisingEnd(const ble::AdvertisingEndEvent &event) override
+    {
+        if (!_is_connecting_or_syncing) {
+            printf("Advertise time elapsed\r\n");
+            next_state();
+        }
+    }
+
+    /** Look at scan payload to find a peer device and connect to it */
+    void onAdvertisingReport(const ble::AdvertisingReportEvent &event) override
+    {
+        /* don't bother with analysing scan result if we're already connecting */
+        if (_is_connecting_or_syncing) {
+            return;
+        }
+
+        /* if we're looking for periodic advertising don't bother unless it's present */
+        if (_is_periodic && !event.isPeriodicIntervalPresent()) {
+            return;
+        }
+
+        ble::AdvertisingDataParser adv_parser(event.getPayload());
+
+        /* parse the advertising payload, looking for a discoverable device */
+        while (adv_parser.hasNext()) {
+            ble::AdvertisingDataParser::element_t field = adv_parser.next();
+
+            /* identify peer by name */
+            if (field.type == ble::adv_data_type_t::COMPLETE_LOCAL_NAME &&
+                field.value.size() == strlen(DEVICE_NAME) &&
+                (memcmp(field.value.data(), DEVICE_NAME, field.value.size()) == 0)) {
+                /* if we haven't established our roles connect, otherwise sync with advertising */
+                if (_is_periodic) {
+                    printf(
+                        "We found the peer, syncing with SID %d and periodic interval %dms\r\n",
```

Comment:
```suggestion
                        "We found the peer, syncing with SID %d and periodic interval %lums\r\n",
```
or `PRIu64` perhaps?